src/utils/SupersetUtils.js

```diff
@@ -0,0 +1,21 @@
+export const buildSupersetDynamicFilters = (reportConfig, context, contextType = 'scenario') => {
+  const { currentScenarioData, visibleScenarios } = context;
+  if (!reportConfig?.dynamicFilters) return [];
+
+  return reportConfig.dynamicFilters.map((filter) => {
+    let values;
+
+    if (contextType === 'scenario') {
+      if (filter.values === 'csmSimulationRun') values = [currentScenarioData?.lastRun?.csmSimulationRun];
+      else if (filter.values === 'lastRunId') values = [currentScenarioData?.lastRunId];
+    }
+
+    if (contextType === 'dashboards') {
+      if (filter.values === 'visibleScenariosCsmSimulationRunsIds')
+        values = visibleScenarios?.map((s) => s.lastRun?.csmSimulationRun);
+      else if (filter.values === 'visibleScenariosSimulationRunsIds') values = visibleScenarios?.map((s) => s.runId);
+    }
+
+    return { table: filter.table, column: filter.column, values: values?.filter(Boolean) ?? [] };
```

Comment:
This structure shape is used in the workspace configuration file, but we probably need to build something different for the filter values that will be forwarded to the Superset component. Let's discuss this 🙂